_includes/gha-tutorial.md

```diff
@@ -0,0 +1,178 @@
+
+To follow along with this guide, you'll need a Docker ID and a GitHub account.
+
+### Create the repository
+
+Create a GitHub repository and configure the Docker Hub secrets.
+
+1. Create a new GitHub repository using
+   [this template repository](https://github.com/dvdksn/clockbox/generate).
+
+   The repository contains a simple Dockerfile, and nothing else. Feel free to
+   use another repository containing a working Dockerfile if you prefer.
+
+2. Open the repository **Settings**, and go to **Secrets** > **Actions**.
+
+3. Create a new secret named `DOCKER_HUB_USERNAME` and your Docker ID as value.
+
+4. Create a new
+   [Personal Access Token (PAT)](https://docs.docker.com/docker-hub/access-tokens/#create-an-access-token)
+   for Docker Hub. You can name this token `clockboxci`.
+
+5. Add the PAT as a second secret in your GitHub repository, with the name
+   `DOCKER_HUB_ACCESS_TOKEN`.
+
+With your repository created, and secrets configured, you're now ready for
+action!
+
+### Set up the workflow
+
+Set up your GitHub Actions workflow for building and pushing the image to Docker
+Hub.
+
+1. Go to your repository on GitHub and then select the **Actions** tab.
+2. Select **set up a workflow yourself**.
+
+   This takes you to a page for creating a new GitHub actions workflow file in
+   your repository, under `.github/workflows/main.yml` by default.
+
+3. In the editor window, copy and paste the following YAML configuration.
+
+   ```yaml
+   name: ci
+
+   on:
+     push:
+       branches:
+         - "main"
+
+   jobs:
+     build:
+       runs-on: ubuntu-latest
+   ```
+
+   - `name`: the name of this workflow.
+   - `on.push.branches`: specifies that this workflow should run on every push
+     event for the branches in the list.
+   - `jobs`: creates a job ID (`build`) and declares the type of machine that
+     the job should run on.
+
+For more information about the YAML syntax used here, see
+[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions){:
+target="blank" rel="noopener"}.
+
+### Define the workflow steps
+
+Now the essentials: what steps to run, and in what order to run them.
+
+The following YAML snippet contains a sequence of steps that:
+
+1. Checks out the repository on the build machine.
+2. Signs in to Docker Hub, using the
+   [Docker Login](https://github.com/marketplace/actions/docker-login){:
+   target="blank" rel="noopener"} action and your Docker Hub credentials.
+3. Creates a BuildKit builder instance using the
+   [Docker Setup Buildx](https://github.com/marketplace/actions/docker-setup-buildx){:
+   target="blank" rel="noopener"} action.
+4. Builds the container image and pushes it to the Docker Hub repository, using
+   [Build and push Docker images](https://github.com/marketplace/actions/build-and-push-docker-images){:
+   target="blank" rel="noopener"}.
+
+   The `with` key lists a number of input parameters that configures the step:
```

Comment:
Should this be after you've shown the code block example?

Reply:
🆗